src/run_macros/infer_route.cr

```diff
@@ -31,7 +31,8 @@ class InferRoute
   end
 
   private def all_pieces
-    (namespace_pieces + parent_resource_pieces + resource_pieces).reject(&.== "")
+    resource_pieces = singular? ? singular_resource_pieces : plural_resource_pieces
```

Comment:
Instead of doing this inline, could you extract a `resource_pieces` private method that uses an `if` instead of ternary? I think that might make it a bit easier to read and change in the future

Something like this
```crystal
private def resource_pieces
  if singular?
    singular_pieces
  else
    plural_resources_pieces
  end
end
```

Reply:
Funny, I had this first, then inlined it. Changed!